lib/cocoapods-core/yaml_helper.rb

```diff
@@ -109,9 +109,21 @@ def process_according_to_class(value, hash_keys_hint = nil)
       #
       def process_array(array)
         result = sorted_array(array).map do |array_value|
-          process_according_to_class(array_value)
+          processed = process_according_to_class(array_value)
+          case array_value
+          when Array, Hash
+            if array_value.size > 1
+              processed = processed.gsub(/^.*/).to_a
+              head = processed.shift
+              processed.map { |s| "  #{s}" }.prepend(head).join("\n")
+            else
+              processed
+            end
```

Comment:
It’s not really clear to me, just from the code, what this is doing exactly. Does the 1 test really cover all the bases?